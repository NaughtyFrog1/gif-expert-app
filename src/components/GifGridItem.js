import React from 'react'

const GifGridItem = ({id, title, url}) => {
  return (
    <li className="card">
      <img src={url} alt={title} />
    </li>
  )
}

export default GifGridItem
