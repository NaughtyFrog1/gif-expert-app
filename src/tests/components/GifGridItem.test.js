import { shallow } from 'enzyme'
import GifGridItem from '../../components/GifGridItem'

describe('tests for GifGridItem', () => {
  const data = {
    title: 'Este es el t√≠tulo',
    url: 'https://esta/url/no/existe.com',
  }
  const wrapper = shallow(<GifGridItem {...data} />)

  //* This test should be enough
  test('should render the component properly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('should have the image attributes', () => {
    const img = wrapper.find('img')

    expect(img.prop('src')).toBe(data.url)
    expect(img.prop('alt')).toBe(data.title)
  })

  test('should have the animation classes', () => {
    expect(
      wrapper
        .find('li')
        .prop('className')
        .includes('animate__animated animate__fadeInUp')
    ).toBe(true)
  })
})
