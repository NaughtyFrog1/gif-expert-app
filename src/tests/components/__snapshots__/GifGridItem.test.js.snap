// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for GifGridItem should render the component properly 1`] = `
<li
  className="gif-cards__card animate__animated animate__fadeInUp"
>
  <img
    alt="Este es el tÃ­tulo"
    src="https://esta/url/no/existe.com"
  />
</li>
`;
